projects/packages/forms/src/blocks/contact-form/components/shared/plugin-integration-panel/index.js

```diff
@@ -0,0 +1,173 @@
+			}
+
+			const plugin = installedPlugins.find( p => p.plugin === pluginPath );
+
+			return {
+				isLoading: false,
+				pluginStatus: {
+					isInstalled: !! plugin,
+					isActive: plugin?.status === 'active',
+				},
+			};
+		},
+		[ pluginPath ]
+	);
+
+	const handleButtonClick = useCallback( () => {
+		const func = pluginStatus?.isInstalled ? activatePlugin : installAndActivatePlugin;
+		const arg = pluginStatus?.isInstalled ? pluginPath : pluginSlug;
+		const isActivationCall = func === activatePlugin;
+
+		setIsInstalling( true );
+		setIsActivating( isActivationCall );
+
+		if ( tracksEventName ) {
+			tracks.recordEvent( tracksEventName );
+		}
+
+		func( arg ).finally( () => {
+			invalidateResolution( 'getPlugins' );
+			setIsInstalling( false );
+			setIsActivating( false );
+		} );
+	}, [
+		pluginStatus?.isInstalled,
+		pluginPath,
+		pluginSlug,
+		tracksEventName,
+		setIsInstalling,
+		setIsActivating,
+		invalidateResolution,
+		tracks,
+	] );
+
+	const renderActionButton = () => {
+		const getButtonLabel = () => {
+			if ( isActivating ) {
+				return __( 'Activating…', 'jetpack-forms' );
+			}
+			if ( isInstalling ) {
+				return __( 'Installing…', 'jetpack-forms' );
+			}
+			return pluginStatus?.isInstalled ? activateText : installText;
+		};
+
+		if ( isInstalling || isActivating ) {
+			return (
+				<Button
+					variant="secondary"
+					icon={ <Icon className="jetpack-plugin-integration__spinner-icon" icon="update" /> }
+					disabled
+					aria-label={ getButtonLabel() }
+				>
+					{ getButtonLabel() }
+				</Button>
+			);
+		}
+
+		return (
+			<Button variant="secondary" onClick={ handleButtonClick }>
+				{ getButtonLabel() }
+			</Button>
+		);
+	};
+
+	const renderContent = () => {
+		// Loading State
+		if ( isLoading ) {
+			return (
+				<div className="jetpack-plugin-integration__status">
+					<div>
+						<Spinner />
+						<span>{ __( 'Checking plugin status…', 'jetpack-forms' ) }</span>
+					</div>
+				</div>
+			);
+		}
+
+		// Error State
+		if ( ! pluginStatus || typeof pluginStatus.isInstalled === 'undefined' ) {
+			return (
+				<div className="jetpack-plugin-integration__status">
+					<Icon icon="warning" />
+					<span>
+						{ __( 'Unable to determine plugin status. Please refresh the page.', 'jetpack-forms' ) }
+					</span>
+				</div>
+			);
+		}
+
+		// Not Installed State
+		if ( ! pluginStatus.isInstalled ) {
+			return (
+				<div className="jetpack-plugin-integration__panel">
+					<div className="jetpack-plugin-integration__panel-content">
+						<div>{ description }</div>
+						{ renderActionButton() }
+					</div>
+				</div>
+			);
+		}
+
+		// Installed but Not Active State
+		if ( ! pluginStatus.isActive ) {
+			return (
+				<div className="jetpack-plugin-integration__panel">
+					<div className="jetpack-plugin-integration__panel-content">
+						<div>
+							{ __(
+								"You already have the plugin installed, but it's not activated.",
```

Comment:
We should mention Akismet here.

Reply:
I've updated this to take in a prop for the plugin name, and output it here. In this case, it'll say "You already have Akismet installed, but it's not activated."